File: t/TypeScript/src/lib/commands/Command.ts
import { Message } from "discord.js";

import { Client } from "../../Client";

/**
 * Constructs a new command
 */
export abstract class Command {
	readonly name: string;
	constructor(readonly client: Client, name: string) {
		this.name = name;
	}

	/**
	 * Execute this command
	 * @param message received from Discord
	 * @param commandName name of this command
	 * @param args provided by user input
	 */
	abstract execute(
		message: Message,
		commandName: string,
		...args: string[]
	): Promise<any>;
}

File: k/kotlin/src/main/kotlin/sh/foxboy/api/command/Command.kt
package sh.foxboy.api.command

import net.dv8tion.jda.api.entities.Message


abstract class Command(name: String) {
    val name = name

    /**
     * Executes a command
     *
     * @param message      The message received
     * @param commandLabel command used
     * @param args         The arguments received
     */
    abstract fun execute(message: Message, commandLabel: String, args: Array<String?>?)
}